### DISCUSSION TITLE
Add support for Webfinger to become identity provider for Tailscale

### DISCUSSION THREAD

**Discussion by gdonval** [simple-login/app#1927] [Feature Requests]
Tailscale is a zero config VPN to build secure networks.
They don't manage nor provide identity for you: they need OIDC-compliant providers for that.

Optimally, I would love to see Simple Login being listed as identity provider on there but barred that, they support a [generic authentication mechanism](https://tailscale.com/kb/1240/sso-custom-oidc/) based on `.well-known` addresses. 

As far as I can tell, Simple Login just needs to implement Webfinger (as described in the link) to become usable by Tailscale and this is what I request here.

**Comment by comminutus** [comment]
would love to see this as well

**Comment by RichardJActon** [comment]
This looks pretty simple to do and would be really great to see implemented

**Comment by zipy124** [comment]
Draft PR at #2199 .

**Comment by Billy5804** [comment]
If anyone wants to sign up to Tailscale with a custom domain using SimpleLogin it is already possible to do so.

I was able to sign up to Tailscale with SimpleLogin using an email on a my custom domain, by temporarily having a worker live on Cloudflare served by my domain and configured to use the route /.well-known/webfinger using this code:
```javascript
addEventListener("fetch", event => {
  const requestURL = new URL(event.request.url);
  const resource = requestURL.searchParams.get("resource");

  if (!resource.includes("tailscale@mydomain.com")) {
    event.respondWith(new Response("", { status: 404 }));
    return;
  }

  const responseBody = JSON.stringify({
    subject: resource,
    links: [
      {
        rel: "http://openid.net/specs/connect/1.0/issuer",
        href: "https://app.simplelogin.io"
      }
    ]
  });

  event.respondWith(new Response(responseBody, {
    headers: { 'content-type': 'application/json;charset=UTF-8' },
  }));
});
```


I had to also create an app through SimpleLogin (in dev mode / unpublished) at https://app.simplelogin.io/developer/ for Tailscale which once created will give you the Clinet ID and Client secret needed by Tailscale and you will need to add `https://login.tailscale.com/a/oauth_response` as an Authorized Redirect URI in the app.
On Tailscale out of the available prompts I believe I selected the login prompt, but I could be wrong.
